src/libraries/System.Private.Xml/tests/AllowDefaultResolverContext.cs

```diff
@@ -32,7 +32,7 @@ private static void ClearCachedSwitch()
             // the default, or alternatively separate all such tests into a separate test project
             // where that project contains only those tests that require the switch set.
 
-            Type t = typeof(XmlConvert).Assembly.GetType("System.Xml.LocalAppContextSwitches");
+            Type t = Type.GetType("System.Xml.LocalAppContextSwitches, System.Private.Xml");
```

Comment:
I've been seeing so much of this lately that I'm wondering if we should just implement the necessary data flow for this.

Reply:
:+1: this is my favorite go to way of doing things

Reply:
The sad part is that even if we implement it (which I think we should) it will only work when the app targets net9+. It's not much use for almost every library out there right now. It will take something like 4 years to really be usable by libraries.
And backporting it to .NET 6 is... difficult to say the least.